hooks/pre_validations/05-set_hostname.rb

```diff
@@ -0,0 +1,27 @@
+if app_value(:hostname) != 'none'
+  hostname = app_value(:hostname)
+
+  logger.debug("hostname \"#{hostname}\" 2>&1")
+  unless kafo.noop?
+    result = `hostname "#{hostname}" 2>&1`
+    if $?.exitstatus > 0
+      say "<%= color('Hostname setup failed: #{result}', :bad) %>"
+      kafo.class.exit(101)
+    end
+  end
+
+  logger.debug("Writing hostname (#{hostname}) to /etc/sysconfig/network")
+  unless kafo.noop?
+    filename = '/etc/sysconfig/network'
+    entry = "HOSTNAME=#{hostname}\n"
+    begin
+      text = File.read(filename) 
+      text.gsub!(/HOSTNAME=.*?\n/m, entry)
```

Comment:
nitpick, missing `^` in regexp, also I think it could be easier if you omit `\n` in entry and this regexp and use `$` for end of the line